pkg/controller/handlers/retention/retention.go

```diff
@@ -0,0 +1,46 @@
+package retention
+
+import (
+	"time"
+
+	"github.com/obot-platform/nah/pkg/router"
+	"github.com/obot-platform/obot/logger"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var log = logger.Package()
+
+type Manager struct {
+	c         kclient.Client
+	policy    time.Duration
+	frequency time.Duration
+}
+
+func NewRetentionManager(c kclient.Client, policy, frequency time.Duration) *Manager {
+	log.Infof("retention policy: %s, frequency: %s", policy, frequency)
+
+	return &Manager{
+		c:         c,
+		policy:    policy,
+		frequency: frequency,
+	}
+}
+
+func (m *Manager) Run(req router.Request, resp router.Response) error {
+	thread := req.Object.(*v1.Thread)
+	if thread.Spec.SystemTask {
+		return nil
+	}
+
+	if !thread.Status.LastUsedTime.IsZero() && time.Since(thread.Status.LastUsedTime.Time) > m.policy {
+		log.Infof("retention: deleting thread %s/%s", thread.Namespace, thread.Name)
+		return m.c.Delete(req.Ctx, thread)
+	}
+
+	if m.frequency < 10*time.Hour {
+		resp.RetryAfter(m.frequency)
+	}
```

Comment:
I would imagine something like this. Then you don't need the frequency parameter.
```suggestion
	if since := time.Since(thread.Status.LastUsedTime.Time); m.policy - since < 10*time.Hour {
		resp.RetryAfter(until)
	}
```

Reply:
Fixed. Good call